src/scripts/delete-confirmation.js

```diff
@@ -1,25 +1,92 @@
 function openDeleteModal(param1, param2, param3, param4) {
-  const modal = document.getElementById('deleteConfirmation');
-  modal.dataset.param1 = param1;
-  modal.dataset.param2 = param2;
-  modal.dataset.param3 = param3;
-  modal.dataset.param4 = param4;
-  const bootstrapModal = new bootstrap.Modal(modal);
-  bootstrapModal.show();
+    const modal = document.getElementById('deleteConfirmation');
+    modal.dataset.param1 = param1;
+    modal.dataset.param2 = param2;
+    modal.dataset.param3 = param3;
+    modal.dataset.param4 = param4;
+    const bootstrapModal = new bootstrap.Modal(modal);
+    bootstrapModal.show();
 }
 
 async function deleteApplication() {
-  const modal = document.getElementById('deleteConfirmation');
-  const applicationId = modal.dataset.param1;
-  try {
-    const response = await fetch(`/devportal/applications/${applicationId}`, { method: 'DELETE' });
-    if (response.ok) {
-      await showAlert('Application deleted successfully!', 'success');
-    } else {
-      await showAlert('Failed to delete application. Please try again.', 'error');
+    const modal = document.getElementById('deleteConfirmation');
+    const applicationId = modal.dataset.param1;
+    const messageOverlay = document.getElementById(`message-overlay-${applicationId}`);
+    const trashButton = document.getElementById(`trash-btn-${applicationId}`);
+
+    try {
+        // Show loading state
+        if (trashButton) {
+            trashButton.innerHTML =
+                '<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;Deleting...';
+            trashButton.disabled = true;
+        }
+        const response = await fetch(`/devportal/applications/${applicationId}`, {
+            method: 'DELETE',
+        });
+        if (response.ok) {
+            // remove the application card from the UI
+            const cardWrapper = document.getElementById(`app-card-${applicationId}`);
+            if (cardWrapper) {
+                const colElement = cardWrapper.closest('.col-12');
+                if (colElement) {
+                    colElement.remove();
+                } else {
+                    cardWrapper.remove();
+                }
+            }
+        } else {
+            if (messageOverlay && typeof window.showAppDeleteMessage === 'function') {
+                window.showAppDeleteMessage(messageOverlay, 'Failed to delete application. Please try again.', 'error');
+            }
+            trashButton.disabled = false;
+            trashButton.innerHTML = '<i class="bi bi-trash"></i>';
+        }
+    } catch (error) {
+        if (messageOverlay && typeof window.showAppDeleteMessage === 'function') {
+            window.showAppDeleteMessage(
+                messageOverlay,
+                'An error occurred while deleting the application. Please try again.',
+                'error',
```

Comment:
Will this show the actual error message or just the error text?

Reply:
taking the word error just to decide the front icon